extensions/scarb-execute/src/lib.rs

```diff
@@ -195,6 +194,8 @@ pub fn execute(
         fs::write(air_private_input_path, output_value)?;
     }
 
+    execution_status?.into_result()?;
```

Comment:
Nit:

I would still prefer checking directly
```suggestion
    if let ExecutionStatus::Panic { reason } = execution_status? {
        return Err(anyhow::anyhow!(reason));
    }
```

Reply:
If it's more readable to you, sure